cmd/api/book/orders_test.go

```diff
@@ -0,0 +1,134 @@
+package book_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/books-service/cmd/api/book"
+	bookmock "github.com/books-service/cmd/api/book/mocks"
+	"github.com/google/uuid"
+	"github.com/matryer/is"
+	gomock "go.uber.org/mock/gomock"
+)
+
+func TestCreateOrder(t *testing.T) {
+
+	t.Run("creates an order with a generic user", func(t *testing.T) {
+
+		is := is.New(t)
+		ctrl := gomock.NewController(t)
+		mockRepo := bookmock.NewMockRepository(ctrl)
+		mockNtfy := bookmock.NewMockNotifier(ctrl)
+		mS := book.NewService(mockRepo, mockNtfy, notificationsTimeout)
+
+		someUser := uuid.New()
+
+		mockRepo.EXPECT().CreateOrder(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, o book.Order) (book.Order, error) {
+			is.True(o.OrderID != uuid.Nil)
+			is.True(o.PurchaserID == someUser)
```

Comment:
```suggestion
			is.Equal(o.PurchaserID, someUser)

Reply:
With `Equal` you can compare a full struct too, to avoid having issues with IDs you can set the expected to the actual before comparing.